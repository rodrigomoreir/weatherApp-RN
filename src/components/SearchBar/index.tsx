import React from 'react';
import { Image } from 'react-native';

import icSearch from '../../assets/icons/icSearch.png';

import {
  StyledContainer,
  StyledTextInput,
  StyledIconContainer
} from './styles';

interface Props {
  placeholder: string;
  value: string;
  onChangeText: () => void;
}

const SearchBar = ({ placeholder, value, onChangeText }: Props) => {
  return (
    <StyledContainer>
      <StyledTextInput
        testID='text-input'
        placeholder={placeholder}
        value={value}
        onChangeText={onChangeText}
      />
      <StyledIconContainer testID='pressable-icon'>
        <Image source={icSearch} />
      </StyledIconContainer>
    </StyledContainer>
  );
};

export default SearchBar;
