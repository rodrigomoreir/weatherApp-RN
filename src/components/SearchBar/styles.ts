import styled, { css } from 'styled-components/native';

export const StyledContainer = styled.View`
  border-radius: 15px;
  height: 40px;
  flex-direction: row;
  ${({ theme: { colors } }) => css`
    background-color: grey;
  `}
`;

export const StyledTextInput = styled.TextInput.attrs(() => ({
  placeholderTextColor: 'black'
}))`
  flex: 1;
  padding-left: 20px;
  color: black;
`;

export const StyledIconContainer = styled.TouchableOpacity.attrs({
  activeOpacity: 1
})`
  margin-right: 20px;
  justify-content: center;
  align-items: center;
`;
