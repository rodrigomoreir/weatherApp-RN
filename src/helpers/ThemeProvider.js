import React from 'react';
import { ThemeProvider as DefaultThemeProvider } from 'styled-components/native';
import PropTypes from 'prop-types';

import { defaultTheme } from '../global/styles/theme';


const ThemeProvider = ({ children }) => (
  <DefaultThemeProvider theme={defaultTheme}>{children}</DefaultThemeProvider>
);

ThemeProvider.propTypes = {
  children: PropTypes.node.isRequired
};

export default ThemeProvider;
