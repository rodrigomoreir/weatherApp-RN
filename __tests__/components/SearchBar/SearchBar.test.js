import React from 'react';
import { fireEvent } from '@testing-library/react-native';

import { renderWithTheme, matchSnapshotWithTheme } from '../../../src/helpers/testing'

import SearchBar from '../../../src/components/SearchBar'

describe('Search Bar', () => {
  test('matches snapshot', () => {
    matchSnapshotWithTheme(
      <SearchBar
        placeholder={'Busque uma cidade'}
        value={'test'}
        onChangeText={jest.fn}
      />
    );
  });

  test('renders correctly', () => {
    renderWithTheme(<SearchBar placeholder='Busque aqui' value='test' onChangeText={jest.fn()} />);
  });

  test('value changes on input text', () => {
    const mockFn = jest.fn();
    const { getByTestId } = renderWithTheme(
      <SearchBar placeholder='Busca' value='test' onChangeText={mockFn} />
    );
    fireEvent.changeText(getByTestId('text-input'), 'value');
    expect(mockFn).toHaveBeenCalledWith('value');
  });
});
