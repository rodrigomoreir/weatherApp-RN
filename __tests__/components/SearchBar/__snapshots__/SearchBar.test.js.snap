// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F5F8FA",
        "borderColor": "#808080",
        "borderRadius": 15,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": 40,
      },
    ]
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Busque uma cidade"
    placeholderTextColor="#969CB2"
    style={
      Array [
        Object {
          "color": "#000000",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingLeft": 20,
        },
      ]
    }
    testID="text-input"
    value="test"
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginRight": 20,
        "opacity": 1,
      }
    }
    testID="pressable-icon"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/icons/icSearch.png",
        }
      }
      style={
        Array [
          Object {
            "tintColor": "#808080",
          },
        ]
      }
    />
  </View>
</View>
`;
